### PR TITLE
feat: add 2022 survey questions

### PR DESCRIPTION
**Pull request by sheplu**
A draft of the questions we should be asking in our survey. 

Some questions are the same as last year, others are new and a few are curated based on last year result.

Of the part `What is important for you ? (Multiple choices)` I used some of the choices from last year (especially the ones big the most answers) and added a few based on our cards from the Dublin Collaborator Summit

### CODE REVIEW

**Comment by sheplu** [comment]
@nodejs/next-10 To review for the next meeting #152

**Comment by mcollina** [comment]
I would add those two questions related to ESM:

1. are you using _native_ ESM, e.g. `"type": "module"` or `.mjs`, in your Node.js Development?
2. are you planning to convert your application to native ESM?

cc @GeoffreyBooth

**Comment by GeoffreyBooth** [comment]
> I would add those two questions related to ESM:

What are you hoping to learn from these questions?

I asked similar questions at the June summit and very few people were planning on converting existing applications to ESM but what seemed like every single hand in the room went up (about 50 people) when asked if they would start their _next_ project in ESM. We could add a third question like “Are you planning to use native ESM for your next Node.js project?” or just replace the second question with this. But either way, what are we trying to learn? That ESM adoption is low?

**Comment by sheplu** [comment]
I can see to have a unique question for both next project and migration (as this is really linked)

Adding a question about the blocker points can be interesting but I am also pretty sure we already know the feedbacks, we mostly all have the same (I think - but can be useful to focus the efforts on the more vocals ones first). What do you think @GeoffreyBooth ?

**Comment by GeoffreyBooth** [comment]
> I am also pretty sure we already know the feedbacks, we mostly all have the same

Do we? We knew the blockers 2-3 years ago, but hopefully most of those have been resolved by now. I know that lots of people are waiting for the Loaders API to go stable (which should happen soon) but I’m unaware of any other issues.

**Comment by sheplu** [comment]
Let's add it then :) ESM/CJS was a big discussion in Austin and we talked again about it during the Dublin collab summit so let's see what is the new feedback from the community !

File: surveys/2022-12/questions.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+## Personal question
+
+- Which constituencies do you identify with?
+    - Direct end users: Users who run tools themselves
+    - Application operators: Users who interacts with existingm running applications
+    - Application developers: Frontend tools consumers, backend server authors, tools authors
+    - Library & package authors: Users who write librairies and packages to be included on other applications
+    - Node.js core maintainers: Developers working directly on Node.js, individuals participating in Working Groups
+    - Organization with investments in Node.js (eg: Enterprises, Gouvernement bodies, Startups, Non-profit)
```
Comment (mhdawson) [inline surveys/2022-12/questions.md:11]: We should check if there were some that we added after the last survey, I think there was.

File: surveys/2022-12/questions.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    - 0 - 10+
+- In what kind of companies are you working in?
+    - big tech
+    - company
+    - startup
+    - university
+    - other
+- Are you part/following some working groups?
+    - yes
+    - no
```
Comment (mhdawson) [inline surveys/2022-12/questions.md:28]: Possibly add "If so which ones?"

File: surveys/2022-12/questions.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    - Organization with investments in Node.js (eg: Enterprises, Gouvernement bodies, Startups, Non-profit)
+- Are you part of a constituency not covered ?
+    - yes
+    - no
+- If not covered what are you part with ? (open question)
+- Where are you from? (open question or countries list)
+    - country
+-  How long have you been working with Node.js? (numbers list)
+    - 0 - 10+
+- In what kind of companies are you working in?
```
Comment (sheplu) [inline surveys/2022-12/questions.md:20]: -companies + organizations

File: surveys/2022-12/questions.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    - big tech
+    - company
+    - startup
+    - university
+    - other
```
Comment (sheplu) [inline surveys/2022-12/questions.md:25]: + individuals

File: surveys/2022-12/questions.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    - Internationalization of the documentation
+    - Good understanding of the direction of the project
+    - Predictable and stable release
+    - Innovation and new features 
+    - Consistent and intuitive error handling
+    - Runtime diagnostic tooling
+    - Relevant API in core
+    - Good understanding and process for security and CVE in the ecosystem
+    - Loader hook (to allow native typescript support)
+- What is your use case of Node.js ?
```
[the remark this answers is not in the document]
Comment (sheplu) [reply surveys/2022-12/questions.md:44]: add this link and a question about priorities

**Comment by joyeecheung** [comment]
It would be valuable to have some data of architecture/OS that people are using to run Node.js and how long do they plan to use it too, to help decide when we are looking into dropping support of a certain platform.

File: surveys/2022-12/questions.md [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+- Are you using native ESM, e.g. `"type": "module"` or `.mjs` in your Node.js Development?
+- Are you planning to convert your application to native ESM?
+- Any recurrent issue with Node.js ? (open question)
+- What need do you have not covered ? (open question)
```
[the remark this answers is not in the document]
Comment (mhdawson) [reply surveys/2022-12/questions.md:48]: we should add dnf and yum to the apt-get, brew list.

**Comment by sheplu** [comment]
Everything updated following suggestions

File: surveys/2022-12/questions.md [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+- For those of you wishing to use ESM in an existing application, what have been the pain points or blockers preventing you from doing so (if any)?
+- Any recurrent issue with Node.js ? (open question)
+- What need do you have not covered ? (open question)
+- How do you get your Node.js executables?
+    - With a Node.js version manager: nvm, ns, n, etc.
+    - Using the official installer
+    - Using package managers like npm or yarn and install Node.js as a package
+    - Using a system package manager: apt-get, brew, dnf, yum ...
+    - Other
+- What is your Architecture and OS in which you are running Node.js ? Do you plan to change that in the near future? (open question)
```
Comment (mcollina) [inline surveys/2022-12/questions.md:59]: I think this question would be better as a multiple choice, using the same names from our supported OS/architecture list.

File: surveys/2022-12/questions.md [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+    - Tutorial and examples of core / complexe features
+    - Internationalization of the documentation
+    - Good understanding of the direction of the project
+    - Predictable and stable release
+    - Innovation and new features 
+    - Consistent and intuitive error handling
+    - Runtime diagnostic tooling
+    - Relevant API in core
+    - Good understanding and process for security and CVE in the ecosystem
+    - Loader hook (to allow native typescript support)
```
Comment (ljharb) [inline surveys/2022-12/questions.md:45]:
```suggestion
    - ESM Loader hook (to allow native typescript support)
```
also would this be "native", or merely equivalent to CJS's "register" patterns?

Comment (GeoffreyBooth) [inline surveys/2022-12/questions.md:45]:
```suggestion
    - Hooks for customization, for example to use TypeScript without a separate build step
```

This isn’t specific to ESM.

Comment (ljharb) [reply surveys/2022-12/questions.md:45]: but the capability already exists in CJS.

File: surveys/2022-12/questions.md [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+- What is your use case of Node.js ?
+- Do you see others Technical Priorities ? [Technical Priorities](https://github.com/nodejs/node/blob/main/doc/contributing/technical-priorities.md) (open question)
+- Are you using native ESM, e.g. `"type": "module"` or `.mjs` in your Node.js Development?
```
[the remark this answers is not in the document]
Comment (GeoffreyBooth) [reply surveys/2022-12/questions.md:48]: I think the current wording is clearer than the suggestion.

File: surveys/2022-12/questions.md (hunk 7 above)
Comment (GeoffreyBooth) [reply surveys/2022-12/questions.md:59]: Also I would think that lots of people develop in one OS/arch and deploy in another, like develop in Mac and deploy under Linux/Docker.

File: surveys/2022-12/questions.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (ljharb) [reply surveys/2022-12/questions.md:48]: "type module" is not automatically native ESM, though, and the confusion around that point has caused boundless newcomer confusion.

File: surveys/2022-12/questions.md [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+- Are you part/following some Working Groups? 
+    - yes
+    - no
+- If part/following some Working Groups, which one? (open question)
+
+## Tech questions
+
+- What is important for you ? (Multiple choices)
+    - Good and up to date documentation
+    - Tutorial and examples of core / complexe features
```
Comment (GeoffreyBooth) [inline surveys/2022-12/questions.md:36]:
```suggestion
    - Tutorial and examples of core / complex features
```

File: surveys/2022-12/questions.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (joyeecheung) [reply surveys/2022-12/questions.md:52]: oops, sorry, that should be `nvs`

File: surveys/2022-12/questions.md [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+- Do you see others Technical Priorities ? [Technical Priorities](https://github.com/nodejs/node/blob/main/doc/contributing/technical-priorities.md) (open question)
+- Are you using native ESM, e.g. `"type": "module"` or `.mjs` in your Node.js Development?
+- Are you planning to convert your application to native ESM?
+- For those of you wishing to use ESM in an existing application, what have been the pain points or blockers preventing you from doing so (if any)?
+- Any recurrent issue with Node.js ? (open question)
+- What need do you have not covered ? (open question)
+- How do you get your Node.js executables?
+    - With a Node.js version manager: nvm, ns, n, etc.
+    - Using the official installer
+    - Using package managers like npm or yarn and install Node.js as a package
```
[the remark this answers is not in the document]
Comment (legendecas) [reply surveys/2022-12/questions.md:56]: I'm aware of a private package manager that installs the specified version of Node.js in the projects' settings automatically when it installs dependencies. It is similar to a `.nvmrc` local to a project.

File: surveys/2022-12/questions.md (hunk 7 above)
Comment (sheplu) [reply surveys/2022-12/questions.md:59]: Add the question for both 
- deployed
- developer

Comment (sheplu) [reply surveys/2022-12/questions.md:59]: updated with the two questions. Is the list of architecture supported this one ? Or is there another file containing the list ? https://nodejs.org/api/process.html#processarch 

```
Possible values are: 'arm', 'arm64', 'ia32', 'mips','mipsel', 'ppc', 'ppc64', 's390', 's390x', and 'x64'.
```

File: surveys/2022-12/questions.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (sheplu) [reply surveys/2022-12/questions.md:48]: I don't really have a great way to word that and I don't find the other wording clearer (nor have a better idea). Should this question be splited to differentiate that ?

File: surveys/2022-12/questions.md [hunk 12]
```diff
@@ -0,0 +1,135 @@
...
+
+- What is important to you that is not in this list? (open question)
+
+## Technical Questions
+
+- Are you using native ESM (e.g. `"type": "module"`,  `.mjs` or other solutions) in your Node.js Development?
+    - Yes
+    - No
+
+- Are you planning to convert your application to native ESM?
```
Comment (GeoffreyBooth) [inline surveys/2022-12/questions.md:129]:
```suggestion
- Are you planning to convert an application to native ESM or use native ESM for a future application?
```

At the June 2022 conference I polled the crowd with this question and almost no one was converting but almost everyone was planning to start their _next_ project as native ESM.

File: surveys/2022-12/questions.md [hunk 13]
```diff
@@ -0,0 +1,135 @@
+# Survey December 2022
```
Comment (GeoffreyBooth) [inline surveys/2022-12/questions.md:1]: We should probably get this survey out soon 😄

Comment (sheplu) [reply surveys/2022-12/questions.md:1]: Haha yeah, I may have to update that :) 

a good timeline would be to 
- have it finished and published end of feb/start of march 
-  let it go a "month" until April/mid April
- review the feedback before Vancouver
- discuss about it during the Vancouver event

Comment (mhdawson) [reply surveys/2022-12/questions.md:1]: I think what we have in the doc makes sense. I think asking for help to do the survey from the Foundation makes senses sooner than later and we can agree on the schedule with them (assuming they can help). @joesepi I know you helped with some past surveys, who at the Foundation do you think we should reach out to for help on the survey?

Comment (mhdawson) [inline surveys/2022-12/questions.md:1]:
```suggestion
# Survey April 2023
```

**Comment by mhdawson** [comment]
I'm going to land this and we can PR in any follow on suggestions.

**Pull request merged**